File: app/src/main/java/com/settlex/android/presentation/dashboard/services/model/ServiceDestination.java
package com.settlex.android.presentation.dashboard.services.model;

import android.app.Activity;

import androidx.annotation.IdRes;

public class ServiceDestination {
    private Class<? extends Activity> activity;
    private Integer navDestinationId;

    public ServiceDestination(Class<? extends Activity> activity) {
        this.activity = activity;
    }

    public ServiceDestination(@IdRes int navDestinationId) {
        this.navDestinationId = navDestinationId;
    }

    public Class<? extends Activity> getActivity() {
        return activity;
    }

    public Integer getNavDestinationId() {
        return navDestinationId;
    }

    public boolean isActivity() {
        return activity != null;
    }

    public boolean isFragment() {
        return navDestinationId != null;
    }
}

File: app/src/main/java/com/settlex/android/presentation/dashboard/services/model/ServiceDestination.kt
package com.settlex.android.presentation.dashboard.services.model

import android.app.Activity

data class ServiceDestination(
    val activity: Class<out Activity>? = null,
    val navDestinationId: Int? = null
) {
    val isActivity get() = activity != null
    val isFragment get() = navDestinationId != null
}
